File: java/search/BinarySearch.java
package search;

public class BinarySearch {
    public static void main(String[] args) {
        // Array ordenado
        int[] array = {1, 2, 5, 6, 9, 12, 15, 21, 28};
        
        // Existe
        int target = 15;
        boolean exists = binarySearch(array, target);
        System.out.println("Existe el elemento " + target + "? " + exists);
        
        // No existe
        target = 10;
        exists = binarySearch(array, target);
        System.out.println("Existe el elemento " + target + "? " + exists);
    }
    
    public static boolean binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;
        
        while (left <= right) {
            int mid = left + (right - left) / 2;
            
            if (arr[mid] == target) {
                return true;
            }
            
            if (arr[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        
        return false;
    }
}

File: python/search/binary_search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1
    
    while left <= right:
        mid = (left + right) // 2
        
        if arr[mid] == target:
            return True
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    
    return False

if __name__ == "__main__":
    # Array ordenado
    array = [1, 2, 5, 6, 9, 12, 15, 21, 28]
    
    # Existe
    target = 15
    exists = binary_search(array, target)
    print(f"¿Existe el elemento {target}? {exists}")
    
    # No existe
    target = 10
    exists = binary_search(array, target)
    print(f"¿Existe el elemento {target}? {exists}")
